src/Event.php

```diff
@@ -596,6 +611,22 @@ public function setType(?string $type): void
         $this->type = $type;
     }
 
+    /**
+     * @param string|float|null $startTimestamp The start time of the event
+     */
+    public function setStartTimestamp($startTimestamp): void
+    {
+        $this->startTimestamp = $startTimestamp;
```

Comment:
If we really want to support the union as typehint, we should throw an exception if the variable is not of the expected type